astrovascpy/scipy_petsc_conversions.py

```diff
@@ -274,11 +282,48 @@ def array2PETScVec(v):
     if MPI_RANK != 0:
         istart_loc = [0]
 
-    # local vector
-    vloc = np_zeros(nloc, PETSc.ScalarType)
+    # Initialize destination array on each rank
+    vloc = np_zeros(nloc, dtype=array_type)
 
     # scatter the vector v on all ranks
-    MPI_COMM.Scatterv([v, nloc_loc, istart_loc, _from_numpy_dtype(PETSc.ScalarType)], vloc, root=0)
+    MPI_COMM.Scatterv([v, nloc_loc, istart_loc, _from_numpy_dtype(array_type)], vloc, root=0)
+
+    return vloc, x
+
+
+def distribute_array(v, array_type=None):
+    """
+    Distributes a numpy array on rank 0 to all ranks using PETSc automatic
```

Comment:
"on rank 0 to all rank". Do you mean "from rank 0 to the other ranks" right? It is a broadcast operation then?
It is not clear to me what the function is doing by reading the comment. Could you please rephrase it, possibly highlight it with a tangible example, and try to use the MPI terminology (broadcast, scatter, gather, allgather ...) if it can help.

Reply:
Yes, it is not very clear. Now I have changed the docstring to stress that it is a scatter operation.